p32/src/bank.rs

```diff
@@ -0,0 +1,442 @@
+#[derive(Debug, PartialEq)]
+pub struct User {
+    name: String,
+    credit_line: u64,
+    balance: i64,
+}
+
+trait BalanceAccessor {
+    fn get_balance(&self) -> i64;
+    fn set_balance(&mut self, value: i64);
+}
+
+impl BalanceAccessor for User {
+    fn get_balance(&self) -> i64 {
+        self.balance
+    }
+
+    fn set_balance(&mut self, value: i64) {
+        self.balance = value
+    }
+}
+
+impl User {
+    // type of name?
+    pub fn new(name: String, credit_line: u64, balance: i64) -> Self {
+        Self {
+            name,
+            credit_line,
+            balance,
+        }
+    }
+}
+
+impl MaxCredit for User {
+    fn max_credit(&self) -> u64 {
+        self.balance as u64 + self.credit_line
```

Comment:
Note that `self.balance as u64` will be equal to 2^64 - 1, i.e. to a very big positive number.

Reply:
done